results/pkg/watcher/reconciler/reconciler_test.go

```diff
@@ -0,0 +1,186 @@
+package reconciler
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/experimental/results/pkg/watcher/convert"
+	"github.com/tektoncd/experimental/results/pkg/watcher/reconciler/common"
+	"github.com/tektoncd/experimental/results/pkg/watcher/reconciler/pipelinerun"
+	"github.com/tektoncd/experimental/results/pkg/watcher/reconciler/taskrun"
+	pb "github.com/tektoncd/experimental/results/proto/proto"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	ttesting "github.com/tektoncd/pipeline/pkg/reconciler/testing"
+	"github.com/tektoncd/pipeline/test"
+	"google.golang.org/protobuf/testing/protocmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/configmap"
+)
+
+func TestReconciler(t *testing.T) {
+	var reconcilerTest ReconcilerTest
+
+	testFuncs := map[string]func(*testing.T){
+		"Update PipelineRun to the existed Result": reconcilerTest.testUpdatePipelineRunToTheExistedResult,
+		"Update TaskRun to the existed Result":     reconcilerTest.testUpdateTaskRunToTheExistedResult,
+	}
+
+	for name, testFunc := range testFuncs {
+		t.Run(name, testFunc)
+	}
+}
+
+type ReconcilerTest struct {
+	taskRun *v1beta1.TaskRun
+	trAsset test.Assets
+
+	pipelineRun *v1beta1.PipelineRun
+	prAsset     test.Assets
+	ctx         context.Context
+	client      pb.ResultsClient
+}
+
+func newReconcilerTest(t *testing.T) *ReconcilerTest {
+	client := common.NewResultsClient(t)
+	tr := &v1beta1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "Tekton-Result",
+			Namespace:   "default",
+			Annotations: map[string]string{"demo": "demo"},
+			UID:         "1",
+		},
+	}
+	pr := &v1beta1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "Tekton-Result",
+			Namespace:   "default",
+			Annotations: map[string]string{"demo": "demo"},
+			UID:         "2",
+		},
+	}
+	trAsset, prAsset, ctx := getFakeClients(t, []*v1beta1.TaskRun{tr}, []*v1beta1.PipelineRun{pr}, client)
+	return &ReconcilerTest{
+		taskRun:     tr,
+		trAsset:     trAsset,
+		pipelineRun: pr,
+		prAsset:     prAsset,
+
+		ctx:    ctx,
+		client: client,
+	}
+}
+
+func getFakeClients(t *testing.T, tr []*v1beta1.TaskRun, pr []*v1beta1.PipelineRun, client pb.ResultsClient) (test.Assets, test.Assets, context.Context) {
+	t.Helper()
+	ctx, _ := ttesting.SetupFakeContext(t)
+	d := test.Data{
+		TaskRuns:     tr,
+		PipelineRuns: pr,
+	}
+	clients, _ := test.SeedTestData(t, ctx, d)
+	cmw := configmap.NewInformedWatcher(clients.Kube, "")
+
+	return test.Assets{
+			Controller: taskrun.NewController(ctx, cmw, client),
+			Clients:    clients,
+		}, test.Assets{
+			Controller: pipelinerun.NewController(ctx, cmw, client),
+			Clients:    clients,
+		}, ctx
+}
+
+func (tt *ReconcilerTest) testUpdatePipelineRunToTheExistedResult(t *testing.T) {
+	tt = newReconcilerTest(t)
```

Comment:
It seems strange to self-update this value. Perhaps we should do something like

```suggestion
func TestUpdatePipelineRunToTheExistedResult(t *testing.T) {
	tt := newReconcilerTest(t)
```